agent/consul/leader_peering.go

```diff
@@ -237,7 +237,7 @@ func (s *Server) syncPeeringsAndBlock(ctx context.Context, logger hclog.Logger,
 		}
 	}
 
-	logger.Trace("checking connected streams", "streams", s.peerStreamServer.ConnectedStreams(), "sequence_id", seq)
+	logger.Trace("checking connected streams", "streams", connectedStreams, "sequence_id", seq)
```

Comment:
This was logging against a new fetch of `ConnectedStreams()` rather than the `connectedStreams` var that we actually reconcile against below